### PR TITLE
VSCode extension and LSP migration and release 

### PR DESCRIPTION
**Pull request by zilto**
The [Hamilton VSCode extension](https://marketplace.visualstudio.com/items?itemName=DAGWorks.hamilton-vsc) is available through the VSCode marketplace.

It is powered by the `sf-hamilton-lsp` [published on PyPi](https://pypi.org/project/sf-hamilton-lsp/) and aliases to `sf-hamilton[lsp]`

The documentation has been updated too.

## TODO
- add tests for LSP
- better communicate to the user when the viz should be reloaded
- better communicate when Python dependencies are missing
- add automated documentation for LSP code (i.e., a "Reference" section)
- unify the documentation and the extension walkthrough feature

### CODE REVIEW

File: dev_tools/vscode_extension/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Copyright (c) 2023-2024 Thierry Jean.
```
Comment (skrawcz) [inline dev_tools/vscode_extension/LICENSE:1]:
```suggestion
Copyright (c) 2024 DAGWorks Inc.

```

**Review by skrawcz — COMMENTED** [review]
Seems okay to me. One thing I found.

File: dev_tools/language_server/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+############################################################################
```
Comment (elijahbenizzy) [inline dev_tools/language_server/tests/conftest.py:1]: OK to have this, cause its apache 2.0?

File: dev_tools/language_server/tests/ls_setup.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+import pytest
+from lsprotocol.types import EXIT, INITIALIZE, SHUTDOWN, ClientCapabilities, InitializeParams
+from pygls.server import LanguageServer
+
+RETRIES = 3
+CALL_TIMEOUT = 3
+
+
+def retry_stalled_init_fix_hack():
```
Comment (elijahbenizzy) [inline dev_tools/language_server/tests/ls_setup.py:30]: comments?

**Review by elijahbenizzy — COMMENTED** [review]
Looks good! Some very minor comments. Should merge then we can revisit asynchronously.

File: dev_tools/language_server/tests/conftest.py (hunk 2 above)
Comment (skrawcz) [reply dev_tools/language_server/tests/conftest.py:1]: yes I think so.

**Pull request merged**